src/content/docs/mongodb/replication-and-sharding.mdx

```diff
@@ -2,5 +2,214 @@
+This is particularly important for large-scale applications where the amount of data exceeds the capacity of a single machine.
+
+#### Importance of Replication and Sharding
+
+Replication and Sharding are  essential for scaling MongoDB and ensuring that your application remains available and performant, even under heavy load or in case of server failures.They provide:
+
+1. **Resilience**: The combination of replication and automatic failover ensures that your application can withstand server failures without significant downtime.
+2. **Performance**: Sharding allows you to manage large datasets and high throughput, making it suitable for modern applications that require quick access to vast amounts of data.
+3. **Scalability**: Both features enable you to scale your MongoDB deployments seamlessly, ensuring that your infrastructure can grow with your application’s demands.
+
+
+## Configuring Replica Sets
+
+Replica sets are important in MongoDB that ensures high availability and data redundancy.Here is step-by-step approach setting up and managing replica sets.
+
+### Prerequisites
+
+Before you begin, ensure you have the following:
+- MongoDB installed on all servers that will be part of the replica set.
+- Access to the command line interface (CLI) on each server.
+
+
+### Step 1: Setting up and  Start MongoDB Instances
+
+Start multiple MongoDB instances on different servers or ports, using the `--replSet` option to specify the replica set name. Here’s an example of starting three instances on a single machine for demonstration purposes:
+
+1. **Start three MongoDB Instances on different servers or ports**
+
+```bash
+# Start the primary instance
+mongod --replSet "rs0" --port 27017 --dbpath /data/db1 --bind_ip localhost
+
+# Start the first secondary instance
+mongod --replSet "rs0" --port 27018 --dbpath /data/db2 --bind_ip localhost
+
+# Start the second secondary instance
+mongod --replSet "rs0" --port 27019 --dbpath /data/db3 --bind_ip localhost
+```
+
+2. **Connect to the Primary Node**:
+
+    ```bash
+    mongo --port 27017
+    ```
+
+
+
+3. **Initiate the Replica Set**:
+
+    ```javascript
+    rs.initiate({
+      _id: "rs0",
+      members: [
+        { _id: 0, host: "localhost:27017" },
+        { _id: 1, host: "localhost:27018" },
+        { _id: 2, host: "localhost:27019" }
+      ]
+    });
+    ```
+
+
+4. **Check Replica Set Status**: Use the following command to verify the status of the replica set.
+
+    ```javascript
+    rs.status();
+    ```
+
+
+Replica sets automatically synchronize data from the primary node to the secondary nodes, providing a mechanism for data recovery and availabilit
+
+
+### Step 2. Managing Replica Set Members
+
+1. Adding a New Member
+
+To add a new member to the replica set, use the `rs.add()` command. For example, to add another secondary member:
+
+```javascript
+rs.add("localhost:27020");
+```
+
+2. Removing a member
+To remove a member from the replica set, use the rs.remove() command:
```

Comment:
formatting issue, please make sure that if your code snippet is under a point then that snippet should come under that specific point. Please format it correctly